project-koku/koku
# [COST-5035] - Use json_query for OCP on GCP sql tag matching

## Jira Ticket

[COST-5035](https://issues.redhat.com/browse/COST-5035)

## Description

This change will tweak our special case tag matching during OCP on GCP trino SQL to avoid false positives on generic string matching.

## Testing

Manually run some trino sql:
1. create table  gcp_tag_test (tags varchar, matched_tag varchar);
2. insert into gcp_tag_test (tags, matched_tag) values ('{"environment":"test","namespace":"fake-namespace"}', '');
3. create table ocp_name_test (namespace varchar);
4. insert into ocp_name_test (namespace) values ('test');
5. Try to match the data!: SELECT gcp.tags, ocp.namespace, ((strpos(lower(gcp.tags), 'namespace') != 0 AND strpos(lower(gcp.tags), lower(ocp.namespace)) != 0)) AS match FROM gcp_tag_test AS gcp CROSS JOIN ocp_name_test AS ocp;
                                             	tags                                             | namespace | match
-------------------------------------------------------------------------------------------------------+-----------+-------
 {"environment":"test","namespace":"fake-namespace"} | test              | true  

The above (OLD method) will match fine

Whereas the below (New) method fails to match (expected)

SELECT gcp.tags, ocp.namespace, (json_query(gcp.tags, 'strict $.namespace' OMIT QUOTES) = ocp.namespace) AS match FROM gcp_tag_test AS gcp CROSS JOIN ocp_name_test AS ocp;

 {"environment":"test","namespace":"fake-namespace"} | test              | false

 You could also test a correct match by inserting the following: 
insert into ocp_name_test (namespace) values ('fake-namespace');
Then run the above sql again


## Release Notes

```markdown
* [COST-5035](https://issues.redhat.com/browse/COST-5035) Fix ocp on gcp tag matching SQL
```
